Extension/src/packageManager.ts

```diff
@@ -104,30 +104,27 @@ export class PackageManager {
         tmp.setGracefulCleanup();
     }
 
-    public DownloadPackages(progress: vscode.Progress<{ message?: string; increment?: number }>): Promise<void | null> {
-        return this.GetPackages()
-            .then((packages) => {
-                let count: number = 1;
-                return this.BuildPromiseChain(packages, (pkg): Promise<void> => {
-                    const p: Promise<void> = this.DownloadPackage(pkg);
-                    progress.report({ message: localize("downloading.progress.description", "Downloading {0}", pkg.description), increment: this.GetIncrement(count, packages.length) });
-                    count += 1;
-                    return p;
-                });
-            });
+    public async DownloadPackages(progress: vscode.Progress<{ message?: string; increment?: number }>): Promise<void | null> {
+        const packages: IPackage[] = await this.GetPackages();
+        let count: number = 1;
+        return this.BuildPromiseChain(packages, async (pkg): Promise<void> => {
+            progress.report({ message: localize("downloading.progress.description", "Downloading {0}", pkg.description), increment: this.GetIncrement(count, packages.length) });
+            count += 1;
+            await this.DownloadPackage(pkg);
+            return;
+        });
     }
 
-    public InstallPackages(progress: vscode.Progress<{ message?: string; increment?: number }>): Promise<void | null> {
-        return this.GetPackages()
-            .then((packages) => {
-                let count: number = 1;
-                return this.BuildPromiseChain(packages, (pkg): Promise<void> => {
-                    const p: Promise<void> = this.InstallPackage(pkg);
-                    progress.report({ message: localize("installing.progress.description", "Installing {0}", pkg.description), increment: this.GetIncrement(count, packages.length) });
-                    count += 1;
-                    return p;
-                });
-            });
+    public async InstallPackages(progress: vscode.Progress<{ message?: string; increment?: number }>): Promise<void | null> {
+        const packages: IPackage[] = await this.GetPackages();
+        let count: number = 1;
+        return this.BuildPromiseChain(packages, async (pkg): Promise<void> => {
+            progress.report({ message: localize("installing.progress.description", "Installing {0}", pkg.description), increment: this.GetIncrement(count, packages.length) });
+            count += 1;
+            await this.InstallPackage(pkg);
+            return;
```

Comment:
return is not needed here

Reply:
I changed the way `BuildPromiseChain `was working (I couldn't leave this comment under the relevant comment that you left). Please take a look at `SequentialResolve` function. it sequentially resolves a sequence of promises.